app/Controllers/Http/UsersController.ts

```diff
@@ -1,19 +1,58 @@
 import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import UsersService from 'App/Services/user.service';
+import CreateProfileValidator from 'App/Validators/CreateProfileValidator';
+import EditUserValidator from 'App/Validators/EditUserValidator';
 
 export default class UsersController {
-    public async getUsers(ctx: HttpContextContract): Promise<{
-        id: Number,
-        title: String
-    }[]> {
-        return [
-            {
-                id: 1,
-                title: 'Hello world',
-            },
-            {
-                id: 2,
-                title: 'Hello universe',
-            },
-        ]
+
+    public async editUserDetails({ auth, request, response }: HttpContextContract): Promise<any> {
+
+        const userId = auth.use('api').user?.id;
+        if (!userId) {
+            throw new Error('Id not Found');
+        }
+        const payload = await request.validate(EditUserValidator)
+        const user = await UsersService.editUser(userId, payload);
+        return response.status(200).json({
+            data: user
+        })
+    }
+
+    public async getUserById({ auth, response }: HttpContextContract): Promise<any> {
+
+        const userId = auth.use('api').user?.id;
+        if (!userId) {
+            throw new Error('Id not Found');
+        }
+        const user = await UsersService.getUserById(userId);
```

Comment:
No need of all this code. You could simply return the `auth.user`. If you want to return profile then you could easily leverage the ORM's relationship feature i.e. `preload() OR load()`